plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestCdfWithManySplits.java

```diff
@@ -0,0 +1,141 @@
+import io.trino.testing.containers.Minio;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import static io.trino.plugin.base.util.Closables.closeAllSuppress;
+import static io.trino.plugin.deltalake.DeltaLakeQueryRunner.DELTA_CATALOG;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.testing.containers.Minio.MINIO_ACCESS_KEY;
+import static io.trino.testing.containers.Minio.MINIO_SECRET_KEY;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestCdfWithManySplits
+        extends AbstractTestQueryFramework
+{
+    private static final String SCHEMA = "test_schema";
+
+    private static final String bucketName = "test-bucket-" + randomNameSuffix();
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Minio minio = closeAfterClass(Minio.builder().build());
+        minio.start();
+        minio.createBucket(bucketName);
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(testSessionBuilder()
+                        .setCatalog(DELTA_CATALOG)
+                        .setSchema(SCHEMA)
+                        .build())
+                .setExtraProperties(ImmutableMap.of(
+                        "query.schedule-split-batch-size", "1",
+                        "node-scheduler.max-splits-per-node", "1",
+                        "node-scheduler.min-pending-splits-per-task", "1"))
+                .build();
+        try {
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.createCatalog("tpch", "tpch");
+
+            queryRunner.installPlugin(new DeltaLakePlugin());
+            queryRunner.createCatalog(DELTA_CATALOG, DeltaLakeConnectorFactory.CONNECTOR_NAME, ImmutableMap.<String, String>builder()
+                    .put("hive.metastore", "file")
+                    .put("hive.metastore.catalog.dir", queryRunner.getCoordinator().getBaseDataDir().resolve("file-metastore").toString())
+                    .put("hive.metastore.disable-location-checks", "true")
+                    .put("hive.s3.aws-access-key", MINIO_ACCESS_KEY)
+                    .put("hive.s3.aws-secret-key", MINIO_SECRET_KEY)
+                    .put("hive.s3.endpoint", minio.getMinioAddress())
+                    .put("hive.s3.path-style-access", "true")
+                    .put("delta.enable-non-concurrent-writes", "true")
+                    .put("delta.register-table-procedure.enabled", "true")
+                    .buildOrThrow());
+
+            queryRunner.execute("CREATE SCHEMA " + SCHEMA + " WITH (location = 's3://" + bucketName + "/" + SCHEMA + "')");
+        }
+        catch (Throwable e) {
+            closeAllSuppress(e, queryRunner);
+            throw e;
+        }
+
+        return queryRunner;
+    }
+
+    @Test
+    public void testReadCdfChangesWithMaxSplitsSetToOne()
```

Comment:
`testReadCdfChangesWithMaxSplitsSetToOne` -> `testReadCdfChanges`

the fact that the max splits is set to 1 is actually handled in the query runner builder